source/guides/recipes/dependency-management.md

```diff
@@ -0,0 +1,101 @@
+(dependency-management-niv)=
+# Managing remote sources with niv
+
+The Nix language can be used to describe dependencies between files managed by Nix.
+Nix expressions themselves can depend on remote sources, and there are multiple ways to specify their origin, as shown in [](pinning-nixpkgs).
+
+For more automation around handling remote sources, set up [niv](https://github.com/nmattia/niv/) in your project:
+
+```shell-session
+$ nix-shell -p niv --run "niv init"
+```
+
+This command will generate `nix/sources.json` in the current directory, which is a lock file for dependencies.
+It will also create `nix/sources.nix`, which exposes those dependencies as an attribute set.
+
+:::{note}
+By default, `niv init` will add the latest revision of `nixpkgs-unstable` as a source.
+If you need the latest revision of a specific branch:
+
+```shell-session
+niv init --nixpkgs nixos/nixpkgs --nixpkgs-branch nixos-23.05
+```
+:::
+
+Import the generated `nix/sources.nix` for the top-level argument in the top-level `default.nix` and use it to refer to the Nixpkgs source directory:
+
+```nix
+{ sources ? import ./nix/sources.nix }:
+let
+  pkgs = import sources.nixpkgs {};
+  build = pkgs.hello;
+in {
+  inherit build;
+}
+```
+
+`nix-build` will call the top-level function with the default argument.
+This pattern allows [overriding remote sources](overriding-sources-niv) programmatically.
+
+Add niv to the development environment for your project to have it readily available:
+
+```diff
+ { sources ? import ./nix/sources.nix }:
+ let
+   pkgs = import sources.nixpkgs {};
+   build = pkgs.hello;
+ in {
+   inherit build;
++  shell = pkgs.mkShell {
++    inputsFrom = [ build ];
++    packages = with pkgs; [
++      niv
++    ];
++  };
+ }
+```
+
+See [](./sharing-dependencies) for details.
```

Comment:
````suggestion
Also add a `shell.nix` to enter that environment more conveniently:

```nix
(import ./. {}).shell
```

See [](./sharing-dependencies) for details, and note that here you have to pass an empty attribute set to the imported expression, since `default.nix` now contains a function.
````